app/controllers/match.js

```diff
@@ -1,11 +1,39 @@
 // require list
 const User = require("../models/User");
+//TODO: add real data instead of users[0]
+//This wil be the list of the liked persons of the logged-in user.
+const {listUser} = "";
+
+async function match(req, res, next) {
+    //gets the personid that will be added to the list
+    try {
+        const {personid} = req.query;
+        const users = await User.find();
+        for (let i = 0; i < listUser.length; i++) {

```

Comment:
Have to loop over the users instead of listUser and check if ID matches the ID in the query:

Reply:
Something like this I think:
```suggestion
        for (let i = 0; i < users.length; i++) {
            if (users[i].id === req.query.likedId) {
                console.log("You've already liked this person");
            }
        }
```

Reply:
You first have to set the 'likedId' in the query before you can check anything!